contracts/staking/PoSValidatorManager.sol

```diff
@@ -425,29 +501,36 @@ abstract contract PoSValidatorManager is
         // a weight update using a higher nonce (which implicitly includes the delegation's weight update)
         // to be used to complete delisting for an earlier delegation. This is necessary because the P-Chain
         // is only willing to sign the latest weight update.
-        require(
-            $._delegatorStakes[delegationID].endingNonce <= nonce,
-            "PoSValidatorManager: nonce does not match"
-        );
+        require(delegator.endingNonce <= nonce, "PoSValidatorManager: nonce does not match");
 
         // Ensure the delegator is pending removed. Since anybody can call this function once
         // end delegation has been initialized, we need to make sure that this function is only
         // callable after that has been done.
         require(
-            $._delegatorStakes[delegationID].status == DelegatorStatus.PendingRemoved,
+            delegator.status == DelegatorStatus.PendingRemoved,
             "PoSValidatorManager: delegation not pending added"
         );
 
-        // Update the delegator status
-        $._delegatorStakes[delegationID].status = DelegatorStatus.Completed;
+        uint256 rewards = $._redeemableDelegatorRewards[delegationID];
+        delete $._redeemableDelegatorRewards[delegationID];
 
-        Delegator memory delegator = $._delegatorStakes[delegationID];
-        _unlock(delegator.weight, delegator.owner);
-        // TODO: issue rewards
+        uint256 validatorFees =
+            rewards * $._validatorRequirements[validationID].delegationFeeBips / 10000;
+
+        // Allocate the delegation fees to the validator.
+        $._redeemableValidatorRewards[validationID] += validatorFees;
+
+        // Reward the remaining tokens to the delegator.
+        _reward(delegator.owner, rewards - validatorFees);
+
+        // Unlock the delegator's stake.
+        _unlock(delegator.owner, weightToValue(delegator.weight));
 
         emit DelegationEnded(delegationID, validationID, nonce);
```

Comment:
Should we include the delegator rewards and the delegator fee amount in this event?